docs/threat-model.md

```diff
@@ -0,0 +1,553 @@
+- **Throughput**: Reduction if enough voting nodes are offline, but system is resilient
+- **Detectability**: Attack is highly visible through network monitoring and vote pattern analysis
+- **Resource Cost**: High cost for attacker relative to limited impact on robust voting system
+
+**Assets Affected**: Blockchain Safety, High Throughput
+
+#### T3: Vote Flooding Attack
+**Description**: Malicious nodes flood the network with invalid or duplicate votes to overwhelm voting infrastructure and waste network resources.
+
+**Prerequisites**:
+- Access to network connectivity (minimal barrier)
+- Knowledge of vote message format
+- Basic understanding of voting protocol
+
+**Attack Vector**:
+1. Craft invalid votes (wrong signatures, non-existent EBs, duplicate votes)
+2. Submit high volume of malicious votes through vote diffusion protocol
+3. Target specific nodes or broadcast widely across network
+4. Exploit any amplification in vote propagation mechanisms
+
+**Cost**: VERY LOW - Requires minimal computational resources and network bandwidth
+
+**Impact**:
+- **Resource Consumption**: Forces legitimate nodes to process and validate spam votes
+- **Network Congestion**: Increased bandwidth usage for vote diffusion protocol
+- **Processing Overhead**: CPU cycles wasted on signature verification of invalid votes
+- **Operational**: Potential degradation of node performance, but doesn't break consensus
+- **Limited Scope**: Pull-based protocols provide natural rate limiting protection
+
+**Assets Affected**: Operational Sustainability, High Throughput
+
+### 3.2 Consensus-Level Threats
+
+#### T4: EB Withholding Attack
+**Description**: Eligible stake pools deliberately withhold EBs they are entitled to create, reducing network throughput and potentially enabling censorship.
+
+**Prerequisites**:
+- Stake pool eligibility for EB creation (via VRF lottery)
+- Economic incentive to withhold (e.g., competing EB producers, censorship goals)
+
+**Attack Vector**:
+1. Win EB creation eligibility through normal VRF process
+2. Either create EB but not propagate it, or simply abstain from creation
+3. May selectively withhold EBs containing specific transactions (censorship)
+4. Could coordinate with other eligible pools to maximize impact
+
+**Cost**: LOW - Opportunity cost of foregone rewards from EB creation
+
+**Impact**:
+- **Throughput**: Reduced transaction processing capacity when EBs are withheld
+- **Censorship**: Potential to delay specific transactions if coordinated
+- **Temporary**: System recovers with next EB opportunity or alternative producers
+- **Limited**: Cannot permanently block transactions due to multiple eligibility opportunities
+
+**Assets Affected**: High Throughput, Decentralization Properties
+
+#### T5: Double Voting
+**Description**: Top 500 voting nodes vote on multiple conflicting EBs that reference the same transactions or create incompatible blockchain states.
```

Comment:
Doesn't the ledger-update protocol sort this out? What, precisely, are "conflicting EBs" aside from EBs that contain mutually incompatible transactions?

Reply:
The description is imprecise here, I'll fix it up on monday